core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -452,37 +490,114 @@ public void unmountException() throws Exception {
-    try {
-      BaseFileSystem.UriCheck(uri);
-    } catch (IllegalArgumentException e) {
-      fail("Should not have thrown an exception with a valid scheme://host:port combination.");
+    for (LoggingEvent e : appender.mEvents) {
+      if (e.getLevel().equals("WARN")) {
+        assertTrue(e.getRenderedMessage().contains("The URI schemes")
+            || e.getRenderedMessage().contains("The URI authority"));
+      }
     }
 
-    uri = new AlluxioURI("/root");
+    Logger.getRootLogger().removeAppender(appender);
+    ConfigurationTestUtils.resetConfiguration();
+  }
+
+  /**
+   * Ensures an exception is thrown when an invalid scheme is passed.
+   */
+  @Test
+  public void uriCheckBadScheme() throws Exception {
+    Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
+    Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
+    AlluxioURI uri = new AlluxioURI("hdfs://localhost:19998/root");
     try {
-      BaseFileSystem.UriCheck(uri);
+      mFileSystem.createDirectory(uri);
+      fail("Should have failed on bad host and port");
     } catch (IllegalArgumentException e) {
-      fail("Should not have thrown an exception when no scheme://host:port supplied.");
+      assertTrue(e.getMessage().contains("Scheme hdfs:// in AlluxioURI is invalid"));
     }
 
-    mFileSystem.createDirectory(new AlluxioURI("alluxio://localhost/root"));
+    ConfigurationTestUtils.resetConfiguration();
+  }
+
+  @Test
+  /**
+   * Ensures sure there are two warnings when a URI with a valid scheme and authority is passed.
+   */
+  public void uriCheckGoodSchemeAuthority() throws Exception {
+    Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
+    Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
+    AlluxioURI uri = new AlluxioURI("alluxio://localhost:19998/root");
+    TestAppender appender = new TestAppender();
+    Logger.getRootLogger().addAppender(appender);
+
+    mFileSystem.createDirectory(uri);
+
+    for (LoggingEvent e : appender.mEvents) {
+      if (e.getLevel().equals("WARN")) {
+        assertTrue(e.getRenderedMessage().contains("The URI schemes")
+            || e.getRenderedMessage().contains("The URI authority"));
+        System.out.println(e.getRenderedMessage());
+      }
+    }
+
+    Logger.getRootLogger().removeAppender(appender);
+    ConfigurationTestUtils.resetConfiguration();
+  }
+
+  /**
+   * Ensures there is no warnings or errors when an {@link AlluxioURI} without a scheme and
+   * authority is passed.
+   */
+  @Test
+  public void uriCheckNoSchemeAuthority() throws Exception {
+    Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
+    Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
+    TestAppender appender = new TestAppender();
+    Logger.getRootLogger().addAppender(appender);
+
+    AlluxioURI uri = new AlluxioURI("/root");
+    mFileSystem.createDirectory(uri);
+
+    if (appender.mEvents.size() > 0) {
```

Comment:
there may be good reasons for unrelated warnings/info messages to be added in the future. Can we make this more narrow so that it only fails if we log the specific warning added here?

Reply:
Sure